src/scrapers/tools/geocoder.py

```diff
@@ -2,58 +2,111 @@
 
-            if geocodeterm.startswith('Stadsdeel'):
-                geocodeterm = geocodeterm + ' Amsterdam'
+        if isinstance(item[1], list):
+            id_array = item[1]
+        else:
+            id_array = item[1].split(',')
 
-            # item.location prepareren voor `landelijk`
-            if geocodeterm in ['Landelijk', 'Heel Nederland']:
-                geocodeterm = 'Nederland'
-            else:
-                # item.location prepareren voor `postcode`
-                zipwithoutspace = p.findall(item.location)
+        geocodeterm = item[0]
+        # item.location prepareren voor stadsdelen
+        geocodeterm = geocodeterm.replace('Amsterdam Algemeen', 'Amsterdam')
+
+        if geocodeterm.startswith('Stadsdeel'):
+            geocodeterm = geocodeterm + ' Amsterdam'
+
+        # is the item.location National?
+        if geocodeterm in ['Landelijk', 'Heel Nederland']:
+            match_address = 'Nederland'
+            match_lat = None
+            match_lon = None
+            msg = "WARNING: " + geocodeterm + " defined as `National`, " + str(len(id_array)) + \
+                  " entries set to `" + \
+                  match_address + "`"
+        else:
+            # is the item.location Dutch postal code correct?
+            zipwithoutspace = p.findall(item.location)
 
-                if len(zipwithoutspace) > 0:
+            if len(zipwithoutspace) > 0:
 
-                    for hit in zipwithoutspace:
-                        geocodeterm = geocodeterm.replace(hit, hit[:4] + '' + hit[4:])
+                for hit in zipwithoutspace:
+                    geocodeterm = geocodeterm.replace(hit, hit[:4] + '' + hit[4:])
 
-            match = self.geolocator.geocode(geocodeterm)
+            match = self.geo_locator.geocode(geocodeterm, country_codes=['NL'],
+                                             featuretype=featuretype,
+                                             addressdetails=True)
 
             if match is None:
-                print("ERROR  : " + geocodeterm + " niet gevonden")
-                item.osm_address = "Niet gevonden"
+                match_lat = None
+                match_lon = None
+                msg = "WARNING: " + geocodeterm + " not found, " + str(len(id_array)) + \
+                      " entries set to `" + \
+                      match_address + "`"
             else:
-                item.osm_address = match.address
-                # Only write location if not already set.
-                if item.latitude is None:
-                    item.latitude = match.latitude
-                    item.longitude = match.longitude
-                db.session.add(item)
-                db.session.commit()
-                print("SUCCESS: " + match.address)
-            db.session.add(item)
-            db.session.commit()
-            time.sleep(1)  # Sleep so we don't overstretch the nominatim api
+                if 'postcode' in match.raw["address"]:
```

Comment:
Enkele en dubbel quotes staan hier gemixt. In de rest van de codebase gebruiken we zoveel mogelijk dubbele quotes. Find replace hier lijkt me het makkelijkst.

Reply:
ok